### ISSUE TITLE
Retrying POST requests on client timeout

### ISSUE THREAD

**Issue by bramkoot** [sindresorhus/got#786]
#### Describe the bug

 - Node.js version: v8.10 
 - Got version: v9.6.0

When setting a timeout on a POST request, and the timeout is reached, a retry is made even though the server did not close the connection, the client did. I don't expect POST requests to be retried, even if a TIMEOUT is reached because of the client timeout. 

#### Actual behavior

POST requests are retried if the client timeout is reached. 

#### Expected behavior

Do not retry POST requests, except (maybe) when the server closed the connection. 

#### Code to reproduce

invoke.js
```js
const got = require('got');

got.post('http://localhost:4001', {
  timeout: 6000,
})
  .then(resp => {
    console.log('GOT response', resp.body)
  })
  .catch(err => console.error('Some error', err))
```

slow-server.js
```js
var http = require('http');
var fs = require('fs');

let counter = 1

http.createServer(function (req, res) {
  const n = counter++
  console.log(`${n} Incoming req`, req.method, req.url)

  setTimeout(() => {
    console.log(`${n} Finishing request`, req.method, req.url)
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end('Hello world');
  }, 10000)
}).listen(4001);
```

Run slow-server.js and after that run invoke.js. You will see the server is receiving multiple requests even though the server is not the one closing the connection. 

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

If we agree that this is an issue, I'm happy to work on a PR to fix it.

**Comment by szmarczak** [comment]
Instead of using `retries` you need to use `retry` according to the [docs](https://github.com/sindresorhus/got/blob/master/readme.md#retry)

**Comment by bramkoot** [comment]
My mistake, the example shouldn't set the retries. The point is that without setting the rerties, got shouldn't be retrying POST requests (at least not out of the box) if the client timeout is reached.

**Comment by szmarczak** [comment]
> If we agree that this is an issue, I'm happy to work on a PR to fix it.

There's no need to wait. I'll be able to check this in a few hours. You can already send a PR if you want to :)

**Comment by szmarczak** [comment]
You can add `beforeRetry` hook to check if Got is really making a new retry.

**Comment by bramkoot** [comment]
Published a fix [here](https://github.com/bramkoot/got/tree/issue-786-fix-retries-on-post-requests) based on the v9.6.0 tag, but not sure which branch I should make the PR to.

**Comment by bramkoot** [comment]
Really happy to see such a quick fix on v10, but I would still like to propose to use my fix to patch the v9.6 which is still the latest released version AFAIK.
